db/hints/manager.cc

```diff
@@ -527,9 +527,7 @@ bool manager::can_hint_for(ep_key_type ep) const noexcept {
 bool manager::check_dc_for(ep_key_type ep) const noexcept {
     try {
         // If target's DC is not a "hintable" DCs - don't hint.
-        // If there is an end point manager then DC has already been checked and found to be ok.
-        return _hinted_dcs.empty() || have_ep_manager(ep) ||
-               _hinted_dcs.find(_local_snitch_ptr->get_datacenter(ep)) != _hinted_dcs.end();
+        return _host_filter.can_hint_for(_local_snitch_ptr, ep);
```

Comment:
We seriously need a node_ptr and datacenter_ptr thing so we don't have to keep looking up hash tables (general remark, not a requirement here).

Reply:
@avikivity what is `node_ptr`?
You can't cache datacenter_ptr here because ep is a parameter here.
The whole idea behind these lines is, first, to look up the corresponding DC in the snitch, and then lookup the corresponding DC in the hints configuration (hints_filter).

Reply:
@piodul Dropping a `have_ep_manager(ep)` check is not a good idea.
This is meant to eliminate the need for those hash tables lookups every time except for the first time we see this `ep` IF this `ep` can be hinted.

One thing that can be optimized here however is that you can also cache negative results in a local cache (`unordered_set<inet_address>) of `inet_address` (which is effectively an int or a bunch of ints for IPv6) thereby optimizing the two string cache lookup into a single `int` cache lookup which should be a lot more efficient.

Reply:
@vladzcloudius The v3 brings back the check, so marking as resolved.